src/api/app/models/package.rb

```diff
@@ -1194,7 +1187,9 @@ def enable_for_repository(repo_name)
   end
 
   def self.is_binary_file?(filename)
-    BINARY_EXTENSIONS.include?(File.extname(filename).downcase)
+    return false unless (mime = Marcel::Magic.by_path(filename))
+
+    !mime.text?
```

Comment:
This needs specs before changing. As we have no idea what is changing otherwise.